smart-web-server/src/main/java/org/smartdata/server/security/RangerAuthorizationManager.java

```diff
@@ -17,31 +17,94 @@
+          Function<RequestAuthorizationContext, RangerOperationDto>>>() {{
+        //cluster
+        put(Pattern.compile("^/api/v2/cluster/nodes$"), ImmutableMap.of(HttpMethod.GET,
+            request -> CLUSTER.getRangerOperationDto(SsmResourceAccessType.VIEW,
+                ALL_VALUES_EXPRESSION)));
+        //rule
+        put(Pattern.compile("^/api/v2/rules$"), ImmutableMap.of(
+            HttpMethod.POST,
+            request -> RULE.getRangerOperationDto(SsmResourceAccessType.CREATE,
+                ALL_VALUES_EXPRESSION),
+            HttpMethod.GET,
+            request -> RULE.getRangerOperationDto(SsmResourceAccessType.VIEW,
+                ALL_VALUES_EXPRESSION)));
+        put(Pattern.compile("^/api/v2/rules/\\d+$"), ImmutableMap.of(
+            HttpMethod.GET,
+            request -> RULE.getRangerOperationDto(SsmResourceAccessType.VIEW,
+                ALL_VALUES_EXPRESSION),
+            HttpMethod.DELETE,
+            request -> RULE.getRangerOperationDto(SsmResourceAccessType.DELETE,
+                ALL_VALUES_EXPRESSION)));
+        //action
+        put(Pattern.compile("^/api/v2/actions$"), ImmutableMap.of(
+            HttpMethod.POST,
+            request -> ACTION.getRangerOperationDto(SsmResourceAccessType.SUBMIT,
+                ALL_VALUES_EXPRESSION),
+            HttpMethod.GET,
+            request -> ACTION.getRangerOperationDto(SsmResourceAccessType.VIEW,
+                ALL_VALUES_EXPRESSION)));
+        put(Pattern.compile("^/api/v2/actions/\\d+$"), ImmutableMap.of(
+            HttpMethod.GET,
+            request -> ACTION.getRangerOperationDto(SsmResourceAccessType.VIEW,
+                ALL_VALUES_EXPRESSION)
+        ));
+        //audit
+        put(Pattern.compile("^/api/v2/audit/events$"), ImmutableMap.of(HttpMethod.GET,
+            request -> AUDIT.getRangerOperationDto(SsmResourceAccessType.VIEW,
+                ALL_VALUES_EXPRESSION)));
+      }};
+
   private final RangerSsmAuthorizer rangerSsmAuthorizer;
 
   @Override
   public AuthorizationDecision check(Supplier<Authentication> authentication,
                                      RequestAuthorizationContext request) {
-
-    return new AuthorizationDecision(rangerSsmAuthorizer.authorize(
-        new RangerAuthorizeRequest(authentication.get().getName(),
-            Collections.emptyList(),
-            request.getRequest().getServletPath(),
-            request.getRequest().getMethod())));
+    return new AuthorizationDecision(
+        OPERATION_MAP.entrySet().stream()
+            .filter(entry -> entry.getKey().matcher(request.getRequest().getServletPath()).find())
+            .map(entry -> Optional.ofNullable(entry.getValue().get(
+                    request.getRequest().getMethod()))
+                .map(func -> rangerSsmAuthorizer.authorize(
+                    new RangerAuthorizeRequest(authentication.get().getName(),
+                        func.apply(request))))
+                .orElse(true))
+            .findAny()
```

Comment:
Maybe better to return in .map() Optional, then filter(Optional::isPresent), .map(Optional::get) and use findAny()

Reply:
done